### ISSUE TITLE
TypeError: Can't instantiate abstract class ChatModelArgs with abstract methods close_server, make_chat_model, prepare_server

### ISSUE THREAD

**Issue by xhluca** [ServiceNow/AgentLab#9]
I did a fresh install of agentlab with `pip install -e .` Running it right away resulted in this error:

```
~/dev/agentlab-internal$ python src/agentlab/ui_assistant.py 
None of PyTorch, TensorFlow >= 2.0, or Flax have been found. Models won't be available and only tokenizers, configuration and file/data utilities can be used.
INFO: $UI_COPILOT_RESULTS_DIR is not defined, Using $HOME/agentlab_results.
Traceback (most recent call last):
  File "/home/xhlu/dev/agentlab-internal/src/agentlab/ui_assistant.py", line 17, in <module>
    chat_model_args=ChatModelArgs(
TypeError: Can't instantiate abstract class ChatModelArgs with abstract methods close_server, make_chat_model, prepare_server
```

Here are the versions:
```
Name: agentlab
Version: 0.0.1
Summary: Main package for developing agents and experiments
Home-page: 
Author: 
Author-email: Rim Assouel <rim.assouel@gmail.com>, Alex Drouin <alex.drouin@servicenow.com>, Maxime Gasse <maxime.gasse@servicenow.com>, Alex Lacoste <alex.lacoste@servicenow.com>, Tom Marty <tom.marty@polymtl.ca>, Massimo Caccia <massimo.caccia1@servicenow.com>
License: BSD-3-Clause
Location: /home/xhlu/dev/agentlab-internal/venv/lib/python3.10/site-packages
Editable project location: /home/xhlu/dev/agentlab-internal
Requires: browsergym, contexttimer, huggingface-hub, ipython, joblib, langchain, langchain-community, langchain-openai, openai, pandas, pyyaml, tiktoken
Required-by: 
---
Name: browsergym
Version: 0.4.2
Summary: BrowserGym: a gym environment for web task automation in the Chromium browser
Home-page: 
Author: Rim Assouel, Léo Boisvert, Massimo Caccia, Alex Drouin, Maxime Gasse, Alex Lacoste, Tom Marty
Author-email: 
License: Apache-2.0
Location: /home/xhlu/dev/agentlab-internal/venv/lib/python3.10/site-packages
Requires: browsergym-core, browsergym-experiments, browsergym-miniwob, browsergym-visualwebarena, browsergym-webarena, browsergym-workarena
Required-by: agentlab
```

**Comment by xhluca** [comment]
Separate issue: changing `ChatModelArgs` to `OpenAIChatModelArgs` will fix it, but we have another error:

```
$ python src/agentlab/ui_assistant.py 
None of PyTorch, TensorFlow >= 2.0, or Flax have been found. Models won't be available and only tokenizers, configuration and file/data utilities can be used.
INFO: $UI_COPILOT_RESULTS_DIR is not defined, Using $HOME/agentlab_results.
Traceback (most recent call last):
  File "/home/xhlu/dev/agentlab-internal/src/agentlab/ui_assistant.py", line 23, in <module>
    flags=Flags(
TypeError: Flags() takes no arguments
```

**Comment by xhluca** [comment]
If i change to `ObsFlag` I get this error instead:

```
$ python src/agentlab/ui_assistant.py 
None of PyTorch, TensorFlow >= 2.0, or Flax have been found. Models won't be available and only tokenizers, configuration and file/data utilities can be used.
INFO: $UI_COPILOT_RESULTS_DIR is not defined, Using $HOME/agentlab_results.
Traceback (most recent call last):
  File "/home/xhlu/dev/agentlab-internal/src/agentlab/ui_assistant.py", line 23, in <module>
    flags=ObsFlags(
TypeError: ObsFlags.__init__() got an unexpected keyword argument 'use_plan'
```

**Comment by xhluca** [comment]
THank you! Will check it out soon.

@kazemnejad FYI
